src/Storage/TSStorageManager.m

```diff
@@ -378,20 +442,29 @@ - (void)deleteThreadsAndMessages {
     [TSAttachmentStream deleteAttachments];
 }
 
-- (void)wipeSignalStorage {
-    self.database = nil;
-    NSError *error;
-
+- (void)deletePasswordFromKeychain {
     [SAMKeychain deletePasswordForService:keychainService account:keychainDBPassAccount];
-    [[NSFileManager defaultManager] removeItemAtPath:[self dbPath] error:&error];
-
+    [[NSUserDefaults standardUserDefaults] removeObjectForKey:TSStorageManagerHasDatabasePassword];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
 
+- (void)deleteDatabaseFile {
+    NSError *error;
+    [[NSFileManager defaultManager] removeItemAtPath:[self dbPath] error:&error];
     if (error) {
         DDLogError(@"Failed to delete database: %@", error.description);
     }
+}
 
+- (void)wipeSignalStorage {
+    self.database = nil;
+    
+    [self deletePasswordFromKeychain];
+    
+    [self deleteDatabaseFile];
+    
     [TSAttachmentStream deleteAttachments];
-
+    
     [[self init] setupDatabase];
```

Comment:
I don't even think this `setupDatabase` bit is necessary, so maybe you *could* use this in multiple places above if you just remove this line.

It's unnecessary, since (IIRC) the only place this is called, we immediately kill the app.  In which case the database would be setup on next launch. (Users access this functionality in Settings > Delete Account)

Though *should* we be killing the app is a different question. Probably not.